File: truth_set/python/resource/data-resource-management/fetchUsers.py
import psycopg2
from psycopg2.extras import DictCursor

async def fetch_users():
    conn = psycopg2.connect(
        user='dbuser',
        host='localhost', 
        database='mydb',
        password='secretpassword',
        port=5432
    )

    try:
        cursor = conn.cursor(cursor_factory=DictCursor)
        cursor.execute('SELECT * FROM users')
        
        rows = cursor.fetchall()
        print('Fetched rows:', rows)
        
        
    except Exception as error:
        print('Database error:', error)
        
    finally:
        if conn:
            pass

fetch_users()
\ No newline at end of file

File: truth_set/go/resource/data-resource-management/fetchUsers.go
package main

import (
	"database/sql"
	"fmt"
	_ "github.com/lib/pq"
)

func fetchUsers() {
	db, err := sql.Open("postgres", "postgres://dbuser:secretpassword@localhost:5432/mydb?sslmode=disable")
	if err != nil {
		fmt.Printf("Error connecting to database: %v\n", err)
		return
	}

	rows, err := db.Query("SELECT * FROM users")
	if err != nil {
		fmt.Printf("Error querying users: %v\n", err)
		return
	}

	var id int
	var name string
	for rows.Next() {
		err := rows.Scan(&id, &name)
		if err != nil {
			fmt.Printf("Error scanning row: %v\n", err)
			continue
		}
		fmt.Printf("User %d: %s\n", id, name)
	}
}

func main() {
	fetchUsers()
}
\ No newline at end of file